ANTLR/src/Compiler/Lang.g4

```diff
@@ -19,16 +19,10 @@ stmt : expr SEMICOLON ;
 expr : val                                                                    #value
      | op=NEGATE val                                                          #uneryOp
      | left=expr op=POW right=expr                                            #binaryOp
-     | left=expr op=DIVIDE right=expr                                         #binaryOp
-     | left=expr op=MULTIPLY right=expr                                       #binaryOp
-     | left=expr op=PLUS right=expr                                           #binaryOp
-     | left=expr op=MINUS right=expr                                          #binaryOp
-     | left=expr op=LOGEQ right=expr                                          #binaryOp
-     | left=expr op=LOGNOTEQ right=expr                                       #binaryOp
-     | left=expr op=LOGLESS right=expr                                        #binaryOp
-     | left=expr op=LOGGREATER right=expr                                     #binaryOp
-     | left=expr op=LOGLESSOREQ right=expr                                    #binaryOp
-     | left=expr op=LOGGREATEROREQ right=expr                                 #binaryOp
+     | left=expr op=(DIVIDE|MULTIPLY) right=expr                                         #binaryOp
+     | left=expr op=(PLUS|MINUS) right=expr                                           #binaryOp
```

Comment:
Do they have same precedence? 
Do divide and multiply now have same precedence?

I am confused because DIVIDE comes on the left hand side of the | 
but MINUS comes on the right hand side of the |

And I would expect DIVIDE AND MINUS to behave the same, in regards to MULTIPLY AND PLUS.

Does it matter?

Reply:
Every line/statement is same precedence so the pipe just means "this or that" so it can be either of the two. This means that Divide/multiply has same precedence and minus/plus has same precedence. However multiply/divide has higher precedence than plus/minus as the Antlr tool gives the first statement the highest precedence followed by the second to highest and so on
*